doc/src/tutorial/solvers.rst

```diff
@@ -23,89 +23,104 @@ However, there is an even easier way.  In SymPy, any expression not in an
-   mean that the equation has no solutions.  It just means that it couldn't
-   find any.  Often this means that the solutions cannot be represented
-   symbolically.  For example, the equation `x = \cos(x)` has a solution, but
-   it cannot be represented symbolically using standard functions.
 
-   >>> solve(x - cos(x), x)
-   Traceback (most recent call last):
-   ...
-   NotImplementedError: multiple generators [x, exp(I*x)]
-   No algorithms are implemented to solve equation exp(I*x)
+In the ``solveset`` module, the linear system of equations is solved using ``linsolve``.
+In future we would be able to use linsolve directly from ``solveset``. Following
+is an example of the syntax of ``linsolve``. 
 
-   In fact, ``solve`` makes *no guarantees whatsoever* about the completeness
-   of the solutions it finds.  Much of ``solve`` is heuristics, which may find
-   some solutions to an equation or system of equations, but not all of them.
+* List of Equations Form:
 
-``solve`` can also solve systems of equations.  Pass a list of equations and a
-list of variables to solve for.
+	>>> linsolve([x + y + z - 1, x + y + 2*z - 3 ], (x, y, z))
+	{(-y - 1, y, 2)}
 
-    >>> solve([x - y + 2, x + y - 3], [x, y])
-    {x: 1/2, y: 5/2}
-    >>> solve([x*y - 7, x + y - 6], [x, y])
-    [(-√2 + 3, √2 + 3), (√2 + 3, -√2 + 3)]
+* Augmented Matrix Form:
 
-.. note::
+	>>> linsolve(Matrix(([1, 1, 1, 1], [1, 1, 2, 3])), (x, y, z))
+	{(-y - 1, y, 2)}
+
+* A*x = b Form
 
-   The type of the output of ``solve`` when solving systems of equations
-   varies depending on the type of the input.  If you want a consistent
-   interface, pass ``dict=True``.
+	>>> M = Matrix(((1, 1, 1, 1), (1, 1, 2, 3)))
+	>>> system = A, b = M[:, :-1], M[:, -1]
+	>>> linsolve(system, x, y, z)
+	{(-y - 1, y, 2)}
 
-   >>> solve([x - y + 2, x + y - 3], [x, y], dict=True)
-   [{x: 1/2, y: 5/2}]
-   >>> solve([x*y - 7, x + y - 6], [x, y], dict=True)
-   [{x: -√2 + 3, y: √2 + 3}, {x: √2 + 3, y: -√2 + 3}]
+.. note::
+
+   The order of solution corresponds the order of given symbols.
 
 .. _tutorial-roots:
 
-``solve`` reports each solution only once.  To get the solutions of a
+``solveset`` reports each solution only once.  To get the solutions of a
 polynomial including multiplicity use ``roots``.
 
-    >>> solve(x**3 - 6*x**2 + 9*x, x)
-    [0, 3]
+    >>> solveset(x**3 - 6*x**2 + 9*x, x)
+    {0, 3}
     >>> roots(x**3 - 6*x**2 + 9*x, x)
     {0: 1, 3: 2}
 
 The output ``{0: 1, 3: 2}`` of ``roots`` means that ``0`` is a root of
 multiplicity 1 and ``3`` is a root of multiplicity 2.
 
+.. note::
+
+   Currently ``solveset`` is not capable of solving the following types of equations:
+
+   * Non-linear multivariate system
+   * Equations solvable by LambertW (Transcendental equation solver).
+
+   ``solve`` can be used for such cases:
+
+   >>> solve([x*y - 1, x - 2], x, y)
+   [(2, 1/2)]
+   >>> solve(x*exp(x) - 1, x )
+   [LambertW(1)]
```

Comment:
I think we should atleast let the beginners know, that there exist a function named `solve`. Which can be used if `solveset` doesn't works. So that they can refer to the full documentation for advanced usage.